regulation/tree.py

```diff
@@ -84,14 +90,22 @@ def build_reg_tree(root, parent=None, depth=0):
         # node rather than child nodes in their own right.
         if len(children) > 0:
             first_child = children[0]
-            # if it doesn't have a title, doesn't have a marker, and
-            # doesn't have children, it is an intro paragraph.
-            if first_child.find('{eregs}title') is None and \
-                    first_child.get('marker') == '' and \
-                    len(first_child.findall('{eregs}paragraph')) == 0:
-                content = xml_node_text(first_child.find('{eregs}content'))
-                node.text = content.strip()
-                del children[0]
+            # if first_child doesn't have a title and doesn't have a marker
+            # it may be an intro paragraph
+            if first_child.find('{eregs}title') is None and first_child.get('marker') == '':
+                is_intro_para = True
+                # Determine whether this content has any top-level-type tags within
+                if len(first_child) > 0:
+                    for child in first_child[0]:
+                        if child.tag in NON_PARA_SUBELEMENT:
+                            is_intro_para = False
```

Comment:
I wonder if it would be worth it to replace this this `if` and `for` loop and `if` with something along the lines of `filter(lambda child: child.tag in NON_PARA_SUBELEMENT, first_child.getchildren())`.

I don't think it'd reduce readability, and it'd eliminate a few levels of `if/for/if`. I don't feel strongly about this though, so if @hillaryj and @grapesmoker are happy, this is fine.